arch/common/CMakeLists.txt

```diff
@@ -22,6 +22,10 @@ zephyr_library_sources_ifdef(
   multilevel_irq.c
   )
 
+if(CONFIG_MULTI_LEVEL_INTERRUPTS)
+  zephyr_library_sources_ifdef(CONFIG_LEGACY_MULTI_LEVEL_TABLE_GENERATION multilevel_irq_legacy.c)
+endif()
```

Comment:
Due to the `depends on`, the surrounding condition can be removed.
```suggestion zephyr_library_sources_ifdef(CONFIG_LEGACY_MULTI_LEVEL_TABLE_GENERATION multilevel_irq_legacy.c)
```

Reply:
Fixed